src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -295,6 +295,8 @@ else if (config.rpc_interface != null)
     public static void applyConfig(Config config) throws ConfigurationException
     {
         conf = config;
+        
+        InetAddressAndPort.initializeDefaultPort(getSSLStoragePort());
```

Comment:
note that on trunk the default port is `getStoragePort()` so 7000. I'm not sure if this matters.